src/env/node/git/git.ts

```diff
@@ -1588,6 +1598,21 @@ export class Git {
 		return this.git<string>({ cwd: repoPath }, 'reset', '-q', '--', ...pathspecs);
 	}
 
+	async revert(repoPath: string, ...args: string[]) {
+		try {
+			await this.git<string>({ cwd: repoPath }, 'revert', ...args);
+		} catch (ex) {
+			const msg: string = ex?.toString() ?? '';
+			for (const [error, reason] of revertErrorAndReason) {
+				if (error.test(msg) || error.test(ex.stderr ?? '')) {
+					throw new RevertError(reason, ex);
+				}
+			}
+
+			throw new RevertError(RevertErrorReason.Other, ex);
+		}
+	}
```

Comment:
That is being performed at `localGitProvider`. From my understanding, `localGitProvider` is an abstraction from GitLens that uses the local implementation for `git` (the git binary in our case). As such, it should receive the options (business logic) and make the proper adaptations to call the local git implementation (the `git` methods). 
The current implementation keeps the methods in `git` as atomic as possible and separated from our business logic (we could even move the error parsing to `localGitProvider` if necessary). 

If you think it would be clearer to perform that here, I could move that logic here and leave the `revert` at `localGitProvider` with just the git call. Is this what you mean?

```ts
// localGitProvider.ts 
export class LocalGitProvider implements GitProvider, Disposable {
	// ...

	revert(repoPath: string, ref: string, options?: { edit?: boolean }): Promise<void> {
		return this.git.revert(repoPath, ref, options);
	}
}


// git.ts
export class Git {
	// ...
	
	async revert(repoPath: string, ref: string, options?: { edit?: boolean }) {
		const args = [];
		if (options.edit !== undefined) {
			args.push(options.edit ? '--edit' : '--no-edit');
		}
		
		args.push(ref);

		try {
			await this.git<string>({ cwd: repoPath }, 'revert', ...args);
		} catch (ex) {
			const msg: string = ex?.toString() ?? '';
			for (const [error, reason] of revertErrorAndReason) {
				if (error.test(msg) || error.test(ex.stderr ?? '')) {
					throw new RevertError(reason, ex, ref);
				}
			}

			throw new RevertError(RevertErrorReason.Other, ex, ref);
		}
	}
}
```